api/v1alpha1/tunnelbinding_types.go

```diff
@@ -118,6 +119,141 @@ type TunnelBindingStatus struct {
+	Settings AccessConfigSettings `json:"settings"`
+}
+
+type AccessConfigSettings struct {
+	// Authentication settins
+	//+kubebuilder:validation:Optional
+	Authentication AccessConfigAuthentication `json:"authentication"`
+	// Appearance settins
+	//+kubebuilder:validation:Optional
+	Appearance AccessConfigAppearance `json:"appearance"`
+	// Cookie settings
+	//+kubebuilder:validation:Optional
+	Cookies AccessConfigCookies `json:"cookies"`
+	// Additional settings
+	//+kubebuilder:validation:Optional
+	Additional AccessConfigAdditional `json:"additional"`
+}
+
+type AccessConfigAuthentication struct {
+	// The list of identiy providers which application is allowed to use. If empty all idps are allowed
+	//+kubebuilder:validation:Optional
+	AllowedIdps []string `json:"allowedIdps"`
+	// Skip identity provider selection if only one is configured
+	//+kubebuilder:validation:Optional
+	//+kubebuilder:default:=false
+	InstantAuth bool `json:"instantAuth"`
+	// The amount of time that tokens issued for this application will be valid. Must be in the format 300ms or 2h45m. Valid time units are: ns, us (or µs), ms, s, m, h.
+	//+kubebuilder:validation:Optional
+	//+kubebuilder:default:="24h"
+	SessionDuration string `json:"sessionDuration"`
+	// The custom URL a user is redirected to when they are denied access to the application.
+	//+kubebuilder:validation:Optional
+	CustomDenyUrl string `json:"customDenyUrl"`
+	// The custom error message shown to a user when they are denied access to the application.
+	//+kubebuilder:validation:Optional
+	CustomDenyMessage string `json:"customDenyMessage"`
+}
+
+type AccessConfigAppearance struct {
+	// Wether to show app in the launcher. Defaults to true.
+	//+kubebuilder:validation:Optional
+	//+kubebuilder:default:=true
+	AppLauncherVisibility bool `json:"appLauncherVisibility"`
+	// Custom logo url
+	//+kubebuilder:validation:Optional
+	CustomLogo string `json:"customLogo"`
+}
+
+type AccessConfigCookies struct {
+	// Sets the SameSite cookie setting, which provides increased security against CSRF attacks. [none,strict,lax]
+	//+kubebuilder:validation:Optional
+	//+kubebuilder:validation:Enum:="";"none";"strict";"lax"
+	SameSiteAttribute string `json:"sameSiteAttribute"`
+	// Enables the HttpOnly cookie attribute, which increases security against XSS attacks.
+	//+kubebuilder:validation:Optional
+	//+kubebuilder:default:=true
+	EnableHttpOnly bool `json:"enableHttpOnly"`
+	// Enables the binding cookie, which increases security against compromised authorization tokens and CSRF attacks.
+	//+kubebuilder:validation:Optional
+	//+kubebuilder:default:=false
+	EnableBindingCookie bool `json:"enableBindingCookie"`
+}
+
+type AccessConfigAdditional struct {
+	// Cloudflare will render an SSH terminal or VNC session for this application in a web browser. [ssh,vnc]
+	//+kubebuilder:validation:Optional
+	//+kubebuilder:validation:Enum:="";"vnc";"ssh"
+	BrowserRendering string `json:"browserRendering"`
+}
```

Comment:
This might need to be validated against the tunnelBinding subject's protocol, or rather use that to determine instead of being a config.